lms/djangoapps/learner_home/test_serializers.py

```diff
@@ -653,26 +646,18 @@ def generate_test_programs_info(cls):
             ],
         }
 
-    def test_happy_path(self):
+    def test_programs_serializer(self):
+        """Test the ProgramsSerializer"""
+        # Given a program with random test data
         input_data = self.generate_test_programs_info()
-        output_data = ProgramsSerializer(input_data).data
 
-        related_programs = output_data.pop("relatedPrograms")
-
-        for i, related_program in enumerate(related_programs):
-            input_program = input_data["relatedPrograms"][i]
-            assert related_program == {
-                "bannerUrl": input_program["bannerUrl"],
-                "estimatedNumberOfWeeks": input_program["estimatedNumberOfWeeks"],
-                "logoUrl": input_program["logoUrl"],
-                "numberOfCourses": input_program["numberOfCourses"],
-                "programType": input_program["programType"],
-                "programUrl": input_program["programUrl"],
-                "provider": input_program["provider"],
-                "title": input_program["title"],
-            }
+        # When I serialize the program
+        output_data = ProgramsSerializer(input_data).data
 
-        self.assertDictEqual(output_data, {})
+        # Then the output should match the serialized input
+        self.assertEqual(output_data, {
+            "relatedPrograms": RelatedProgramSerializer(input_data["relatedPrograms"], many=True).data
+        })
```

Comment:
This only tests that you're using the right serializer. Need to add tests for the `RelatedProgramSerializer` to test the output of that.